src/algebra/lie_algebra.lean

```diff
@@ -612,20 +634,83 @@ end quotient
 
 end lie_submodule
 
-/--
-An important class of Lie rings are those arising from the associative algebra structure on
-square matrices over a commutative ring.
--/
-def matrix.lie_ring (n : Type u) (R : Type v)
-  [fintype n] [decidable_eq n] [comm_ring R] : lie_ring (matrix n n R) :=
+section matrices
+open_locale matrix
+
+variables {R : Type u} [comm_ring R]
+variables {n : Type w} [fintype n] [decidable_eq n]
+
+/-- An important class of Lie rings are those arising from the associative algebra structure on
+square matrices over a commutative ring. -/
+def matrix.lie_ring : lie_ring (matrix n n R) :=
```

Comment:
You switched from explicit to implicit variables. I'm not sure if you want that.

Reply:
This was a deliberate change, though I confess I can't think of a convincing reason for either explicit or implicit. So far all uses of these defs are by making them local class instances where it doesn't matter.

My reasoning was not more than that I definitely did want implicit variables for `lie_equiv_matrix'` and it looked worth sharing variables in a section with the `matrix.lie_ring`, `matrix.lie_algebra` defs. So I just declared they should all be implicit.

Do you think I should keep explicit variables for `matrix.lie_ring`, `matrix.lie_algebra`?

Reply:
Like you say it doesn't matter too much. So, if it was deliberate, it's fine with me.